seacrowd/sea_datasets/generated_review_enth/generated_review_enth.py

```diff
@@ -0,0 +1,152 @@
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="generated_review_enth_source",
+            version=SOURCE_VERSION,
+            description="Generated Review EN-TH source schema",
+            schema="source",
+            subset_id="generated_review_enth",
+        ),
+        SEACrowdConfig(
+            name="generated_review_enth_seacrowd_t2t",
+            version=SEACROWD_VERSION,
+            description="Generated Review EN-TH SEACrowd schema",
+            schema="seacrowd_t2t",
+            subset_id="generated_review_enth",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "generated_review_enth_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "en_segment": datasets.Value("string"),
+                    "th_segment": datasets.Value("string"),
+                    "review_star": datasets.Value("int32"),
+                    "correct": datasets.Value("bool"),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(
+        self, dl_manager: datasets.DownloadManager
+    ) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+        data_dir = os.path.join(data_dir, "data")
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train.jsonl"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "test.jsonl"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "valid.jsonl"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
```

Comment:
If you're not using the `split` keyword argument, I suggest removing it (and also in the `SplitGenerator`). The one on the template is just a guide on how it's used.